waddrmgr/scoped_manager.go

```diff
@@ -579,6 +593,75 @@ func (s *ScopedKeyManager) AccountProperties(ns walletdb.ReadBucket,
 	return props, nil
 }
 
+// cachedKey is an entry within the LRU map that stores private keys that are
+// to be used frequently. We use this wrapper struct to be able too report the
+// size of a given element to the cache.
+type cachedKey struct {
+	key *btcec.PrivateKey
+}
+
+// Size returns the size of this element. Rather than have the cache limit
+// based on bytes, we simply report that each element is of size 1, meaning we
+// can set our cached based on the amount of keys we want to store, rather than
+// the total size of all the keys.
+func (c *cachedKey) Size() (uint64, error) {
+	return 1, nil
+}
+
+// DeriveFromKeyPathCache is identical to DeriveFromKeyPath, however it'll fail
+// if the account refracted in the DerivationPath isn't already in the
+// in-memory cache. Callers looking for faster private key retrieval can opt to
+// call this method, which may fail if things aren't in the cache, then fall
+// back to the normal variant.
+func (s *ScopedKeyManager) DeriveFromKeyPathCache(kp DerivationPath,
+) (*btcec.PrivateKey, error) {
+
+	s.mtx.Lock()
+	defer s.mtx.Unlock()
+
+	// First, try to look up the key itself in the proper cache, if the key
+	// is here, then we don't need to do anything further.
+	privKeyVal, err := s.privKeyCache.Get(kp)
```

Comment:
So even though we only use `InternalAccount`, `Branch` and `Index` of the derivation path, we still use the full derivation path as the key here. Likely not a problem if we always use this the same way. So not sure if adding a new struct just for the things we actually need as the key is overkill...

Reply:
> Since btcsuite is supposed to be attached to a single btcd node, would there ever be a situation where we'd need to store pks of different networks (testnet, regtest, mainnet)?

No, since a single instance is only every tracking a single chain. 

> . So not sure if adding a new struct just for the things we actually need as the key is overkill...

You mean a new struct that only has the internal account? We use things pretty consistently on the caller side of things, only setting the `InternalAccount` as that's what's actually used to derive the keys.

Reply:
> You mean a new struct that only has the internal account?

Yes, only the internal account and no `MasterKeyFingerprint`. But as I said, as long as we call the cached method in a consistent manner concerning the fields being set, this shouldn't be a problem.